packages/babel-parser/src/parser/statement.js

```diff
@@ -994,8 +994,16 @@ export default class StatementParser extends ExpressionParser {
     this.next();
     this.takeDecorators(node);
```

Comment:
Do you know if class decorators are also technically auto-strict? The spec currently says

> All parts of a ClassDeclaration or a ClassExpression are strict mode code. 

and I have no idea if that is something that is meant to include decorators, or if the decorators proposal will be changing that line to specifically call out the Body and Heritage.

I could see auto-strict decorators being annoying if support for decorating things other than classes were to be considered in the future.

Reply:
The decorators please doesn't say anything about it. @littledan wdyt about it?

Reply:
What of you directly write the function inside the decorator?

Reply:
I misunderstood the question :-) i believe without additional specification, it’s just like any other function - inherits strict from the lexical context or uses the pragma to turn it on.